### PR TITLE
Allow Elasticsearch to be used as an engine for DMS Endpoint

### LINKED ISSUE #9899: Add support for aws_dms_endpoint elasticsearch target engine type
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In 2018 AWS added support for DMS endpoints supporting Elasticsearch as a target engine type.  It would be great if terraform supported this.

### New or Affected Resource(s)

* aws_dms_endpoint

### Potential Terraform Configuration

```
resource "aws_dms_endpoint" "my-endpoint" {
...
  endpoint_type = "target"
  engine_name   = "elasticsearch"
...
}
```
### References

AWS documentation is here:
* https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Target.Elasticsearch.html

### PR DESCRIPTION
**Pull request by timrcoulson**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #9899

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_dms_endpoint: allow `elasticsearch` to be used for the engine for DMS
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAwsDmsEndpoint_Elasticsearch'
?       github.com/terraform-providers/terraform-provider-aws   [no test files]
=== RUN   TestAccAwsDmsEndpoint_Elasticsearch
=== PAUSE TestAccAwsDmsEndpoint_Elasticsearch
=== CONT  TestAccAwsDmsEndpoint_Elasticsearch
--- PASS: TestAccAwsDmsEndpoint_Elasticsearch (46.01s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       46.047s
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws/internal/flatmap      0.022s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags 0.006s [no tests to run]
```

### CODE REVIEW

File: aws/resource_aws_dms_endpoint.go [hunk 1]
```diff
@@ -217,6 +218,41 @@ func resourceAwsDmsEndpoint() *schema.Resource {
...
+						},
+						"endpoint_uri": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"error_retry_duration": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  "10",
```
Comment (johanwiren) [inline aws/resource_aws_dms_endpoint.go]: As per https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Target.Elasticsearch.html the default should be 300

File: aws/resource_aws_dms_endpoint.go [hunk 2]
```diff
@@ -217,6 +218,41 @@ func resourceAwsDmsEndpoint() *schema.Resource {
...
+						},
+						"full_load_error_percentage": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  "300",
```
Comment (johanwiren) [inline aws/resource_aws_dms_endpoint.go]: And this one should default to 10

File: website/docs/r/dms_endpoint.html.markdown [hunk 3]
```diff
@@ -65,6 +65,7 @@ The following arguments are supported:
...
 * `s3_settings` - (Optional) Settings for the target S3 endpoint. Available settings are `service_access_role_arn`, `external_table_definition`, `csv_row_delimiter` (default: `\\n`), `csv_delimiter` (default: `,`), `bucket_folder`, `bucket_name` and `compression_type` (default: `NONE`). For more details, see [Using Amazon S3 as a Target for AWS Database Migration Service](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Target.S3.html).
+* `elasticsearch_settings` - (Optional) Settings for the target Elasticsearch. Available settings are `service_access_role_arn`, `endpoint_uri`, `error_retry_duration` (default: `10`) and `full_load_error_percentage` (default: `300`). For more details, see [Using an Amazon Elasticsearch Service Cluster as a Target for AWS Database Migration Service](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Target.Elasticsearch.html).
```
Comment (johanwiren) [inline website/docs/r/dms_endpoint.html.markdown]: Default values for error_retry_duration and full_load_error_percentage are swapped.

**Comment by timrcoulson** [comment]
@johanwiren swapped the default values

**Comment by ShaneCal** [comment]
We urgently need support for this. We are planning on streaming data from our production databases to ELK in the next week and I really don't want to put this in through the console

**Pull request merged**